src/utils/downloadImageIfNeeded.ts

```
@@ -0,0 +1,49 @@
+
+    if (!URLObject.pathname.endsWith('.png') && !URLObject.pathname.endsWith('.jpg')) {
+      throw new Error('Only PNG and JPG is supported');
+    }
+
+    if (!['https:', 'http:'].includes(URLObject.protocol)) {
+      throw new Error('image is not served as HTTP/S URL');
+    }
+
+    validUrl = URLObject.href;
+  } catch (error) {
+    throw error;
```

Biome (1.9.4)

[error] The catch clause that only rethrows the original error is useless.

An unnecessary catch clause can be confusing.
Unsafe fix: Remove the try/catch clause.


(lint/complexity/noUselessCatch)